jetstream/statistics.py

```diff
@@ -461,9 +461,19 @@ def transform(
         covariate_col_label = None
         if self.covariate_adjustment is not None:
             covariate_period = parser_metric.AnalysisPeriod(self.covariate_adjustment["period"])
-            if covariate_period != self.period:
+            # we cannot apply covariate adjustment if the adjusting period is the current period
+            # or if the current period is itself a pre-enrollment period (e.g., cannot adjust
+            # preenrollment_week using preenrollment_days28)
+            if (covariate_period != self.period) and self.period not in (
+                parser_metric.AnalysisPeriod.PREENROLLMENT_WEEK,
+                parser_metric.AnalysisPeriod.PREENROLLMENT_DAYS_28,
+            ):
```

Comment:
I wonder if we should have a `PREENROLLMENT_PERIODS` global that we can reuse in the validator above too.. What do you think?

Reply:
That would be nice. Should that live in Jetstream or in metric-config-parser (where this Enum is originally defined)?

Reply:
Oh I just meant in the top of this file.